packages/marshal/src/stringify-capdata.js

```diff
@@ -0,0 +1,57 @@
+/** @template S @typedef {import('./types.js').CapData<S>} CapData */
+
+const { Fail, quote: q } = assert;
+
+/** @typedef {string} SimpleString */
+
+/**
+ * A SimpleString is one where JSON.strigify-ing the string just
+ * puts quotes around the contents, and it contains no close square
+ * brackets. Many slot encodings use slot strings that
+ * obey these rules, which is why they are useful.
+ *
+ * @param {unknown} str
+ * @returns {asserts str is SimpleString}
+ */
+export const assertSimpleString = str => {
+  assert(typeof str === 'string');
+  const stringified = JSON.stringify(str);
+  const expected = `"${str}"`;
+  stringified === expected ||
+    Fail`Expected to stringify to ${q(expected)}, not ${q(stringified)}}`;
+  assert(str.indexOf(']') === -1);
+};
+harden(assertSimpleString);
+
+/**
+ * @param {CapData<SimpleString>} capData
```

Comment:
since `SimpleString = String` this won't give a type error for plain string.

if you want to ensure that `assertSimpleString` was called first, the SimpleString type could have a tag so it's a superset of a regular string: The [`Opaque` utility from type-fest](https://github.com/sindresorhus/type-fest/blob/main/source/opaque.d.ts) makes this easy